### ISSUE TITLE
Can I modify the parameters of the function with uprobe(eBPF)?  It is applied only to the test environment.

### ISSUE THREAD

**Issue by billysir** [iovisor/bcc#3376]
**errors occur as follow:**
bpf: Failed to load program: Permission denied
0: (b7) r2 = 44
1: (7b) *(u64 *)(r1 +0) = r2
invalid bpf_context access off=0 size=8

It seems that I can not modify the registers when the arguments passing by registers, since the verifier would refuse.
I just do this in test  environment. How can I do? thranks.

**Comment by yonghong-song** [comment]
No, you cannot change them. These registers are saved by interrupt handler and will restore to the register states for user space codes to continue after interrupt. There is a helper `bpf_probe_write_user()` you can write to the *current* process virtual memory. But in this case, looks like you want to change *register* values, so the helper doesn't work.

**Comment by billysir** [comment]
> No, you cannot change them. These registers are saved by interrupt handler and will restore to the register states for user space codes to continue after interrupt. There is a helper `bpf_probe_write_user()` you can write to the _current_ process virtual memory. But in this case, looks like you want to change _register_ values, so the helper doesn't work.

**Comment by billysir** [comment]
As I understand it, we cannot change them just because of some restrictions in kernel. 
If we just cosider the implementation, it is possible modify the arguments although passing by registers?

**Comment by chenhengqi** [comment]
FYI, you can change Golang function's parameters on the flight since its calling conventions are stack based.
